docs/user/guides/setup/configuration-file.rst

```diff
@@ -0,0 +1,223 @@
+      object in Python. This page won't explain the structure of YAML files, but many resources exist
+      online.
+
+
+.. _YAML: https://en.wikipedia.org/wiki/YAML
+
+.. _howto_templates:
+
+Getting started with a template
+-------------------------------
+
+Here are some variations of the configuration file that can help you get started.
+Pick the one that resembles your project the most,
+copy its contents to ``.readthedocs.yaml`` and add the file to your Git repository.
+
+.. tabs::
+
+    .. tab:: Sphinx
+
+        If your project uses Sphinx,
+        we offer a special builder optimized for Sphinx projects.
+
+        .. literalinclude:: /config-file/examples/sphinx/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+
+    .. tab:: MkDocs
+
+        If your project uses MkDocs,
+        we offer a special builder optimized for MkDocs projects.
+
+        .. literalinclude:: /config-file/examples/mkdocs/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+    .. tab:: Other tools
+
+        If you are using another documentation tool,
+        you can configure the commands of your documentation tool in the following code.
+
+        You will need to create the output directories and direct your documentation tool to write its outputs into those directories.
+
+        .. literalinclude:: /config-file/examples/custom-commands-echo/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+
+Editing the template
+--------------------
+
+Now that you have a ``.readthedocs.yaml`` file added to your Git repository,
+you should see Read the Docs trying to build your project with the configuration file.
+The configuration file probably needs some adjustments to accommodate exactly your project setup.
+
+.. note::
+
+   If you added the configuration file in a separate branch,
+   you may have to activate a :doc:`version </versions>` for that branch.
+
+   If you have added the file in a pull request,
+   you should enable :doc:`pull request builds </guides/pull-requests>`.
+
+Skip: file header and comments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are some parts of the templates that you can leave in place:
+
+Comments
+  We added a lot of comments that explain stuff. They begin with a ``#``.
+
+Commented out features
+  Similar to comments, we also use the ``#`` in front of some configuration options that may or may not be relevant.
+  To enable something that's commented out,
+  simply delete the ``#`` and a blank space in front of the YAML code that you want to enable.
+
+``version`` key
+  The version key tells the system how to read the rest of the configuration file.
+  The current and only supported version is **version 2**.
+
+
+Adjust: ``build.os``
+~~~~~~~~~~~~~~~~~~~~
+
+In our examples,
+we are using Read the Docs' custom image based on the latest Ubuntu LTS release.
+LTS means long-term support,
+meaning that your builds should not break within next many years.
```

Comment:
Builds not breaking might be a lofty statement :smile: 

LTS might be the wrong focus here, I'm not sure. We're trying to tell the user that they won't need to change anything for several years by selecting an operating system version. We could skip on describing LTS here entirely, as this is more a security concept perhaps, and Ubuntu package freezing is the main benefit.